packages/twenty-server/src/engine/metadata-modules/relation-metadata/relation-metadata.service.ts

```diff
@@ -317,4 +318,42 @@ export class RelationMetadataService extends TypeOrmQueryService<RelationMetadat
 
     return deletedRelationMetadata;
   }
+
+  async getManyRelationMetadataByFieldMetadataIdsForDataloader(
+    fieldMetadataIds: string[],
+  ) {
+    const relationMetadataItems = await this.relationMetadataRepository.find({
+      where: [
+        {
+          fromFieldMetadataId: In(fieldMetadataIds),
+        },
+        {
+          toFieldMetadataId: In(fieldMetadataIds),
+        },
+      ],
+      relations: [
+        'fromObjectMetadata',
+        'toObjectMetadata',
+        'fromFieldMetadata',
+        'toFieldMetadata',
+      ],
+    });
+
+    const mappedResult = fieldMetadataIds.map((fieldMetadataId) => {
+      const foundRelationMetadataItem = relationMetadataItems.find(
+        (relationMetadataItem) =>
+          relationMetadataItem.fromFieldMetadataId === fieldMetadataId ||
+          relationMetadataItem.toFieldMetadataId === fieldMetadataId,
+      );
+
+      return (
+        foundRelationMetadataItem ??
+        new NotFoundError(
+          `RelationMetadata with fieldMetadataId ${fieldMetadataId} not found`,
+        )
+      );
+    });
```

Comment:
Do we really need to map here, we can just check the length of `relationMetadataCollection` against `fieldMetadataIds`, if it's not matching we have a problem an then we throw an error it should be enough